lib/src/lifecycle_module.dart

```diff
@@ -272,6 +287,12 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
   /// [onLoad] call will be emitted.
   Stream<LifecycleModule> get didLoad => _didLoadController.stream;
 
+  /// The [LifecycleModule] has been declared firstUseful
+  ///
+  /// Used for the parent module to know that its child is useful, as oftentimes
+  /// a parent can't be called "Useful" until all of its children are
+  Stream<LifecycleModule> get isFirstUseful => _isFirstUsefulController.stream;
```

Comment:
Should this actually be a stream or just a future?

Reply:
That's a good point. A future is a better solution

Reply:
Do we support unloading and reloading a module? My recollection is that we punted on that a long time ago, although now with Disposable in the mix we probably really don't support that. So realistically this should probably be a future.

Reply:
For sure, I was considering taking a different approach at first but I realized our use case of the LifecycleModule is probably both wrong and unusual so the single-load / future approach is definitely the way to go 😄

Reply:
This might actually be better as a Stream since subscriptions can be cleanly cancelled, unlike with futures.

We'll likely want this, as we can't assume that:
- all traced modules will emit this event
- traced modules will emit this event before the listener is disposed

Also, if it were a stream, we could just make it be an empty stream when the module is not traced and thus will never be called, causing the subscription to immediately be cancelled.

```dart
listenToStream(childModule.isFirstUseful.take(1), (module) {
  // ...
});
```
vs
```dart
childModule.isFirstUseful.then((module) {
  if (iAmDisposedOrThisThenBlockIsOtherwiseInvalid) return;
  // ...
});
```

Reply:
Aha I didn't know about .take(1), that's perfect. Thanks Greg.

Reply:
Can we just expose the `take(1)` stream so consumers don't have to know to do that?

Reply:
Part of me likes the convenience of building that in, but another part of me thinks that it obscures the actual behavior and discourages defensive/robust stream listening practices.

Reply:
Obscures the actual behavior? It's a stream that will never issue more than one event. Isn't that the actual behavior?